core/src/test/java/org/jdbi/v3/core/internal/AnnotationFactoryTest.java

```diff
@@ -25,17 +29,9 @@
 public class AnnotationFactoryTest {
     @Test
     @Foo
-    public void create() throws Exception {
-        Foo synthetic = AnnotationFactory.create(Foo.class);
-        Foo real = getClass()
-            .getMethod("create")
-            .getAnnotation(Foo.class);
-
-        assertThat(real).isEqualTo(synthetic);
-        assertThat(synthetic).isEqualTo(real);
-        assertThat(synthetic.hashCode()).isEqualTo(real.hashCode());
-        assertThat(synthetic.toString()).isEqualTo(real.toString());
-        assertThat(synthetic.annotationType()).isEqualTo(Foo.class);
+    public void createNoProperties() throws Exception {
+        checkImplementation(Foo.class, "createNoProperties", AnnotationFactory.create(Foo.class),
+                "@org.jdbi.v3.core.internal.AnnotationFactoryTest$Foo()");
```

Comment:
While `equals` and `hashCode` are well-specified, `toString` is not, so I check it against expected values rather than "whatever the current JVM does".